edx_exams/apps/core/management/commands/bulk_add_course_staff.py

```diff
@@ -0,0 +1,72 @@
+"""Management command to populate the CourseStaffRole model and related User objects, from LMS"""
+import logging
+
+import unicodecsv
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from edx_exams.apps.core.models import (
+    CourseStaffRole,
+    User,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    todo: add docs here
+    """
+    help = 'todo: Add help here'
+
+    def add_arguments(self, parser):
+        # todo: should csv path be required?
+        parser.add_argument(
+            '-p', '--csv_path',
+            metavar='csv_path',
+            dest='csv_path',
+            required=True,
+            help='Path to CSV file.')
+
+    @transaction.atomic
+    def handle(self, *args, **options):
+        """
+        todo: Main logic
+        """
+        csv_path = options['csv_path']
+
+        with csv_path:
+            with open(csv_path, 'rb') as csv_file:
+                self.add_course_staff_from_csv(csv_file)
+
+        logger.info('Bulk add course staff complete!')
+
+    def add_course_staff_from_csv(self, csv_file):
+        """
+        Add the given set of course staff provided in csv
+        """
+        reader = list(unicodecsv.DictReader(csv_file))
+        for row in reader:
+            self.add_course_staff(row.get('username'), row.get('email'), row.get('role'))
+
+    def add_course_staff(self, username, email, role):
+        # todo: add error handling
+        user, _ = User.objects.get_or_create(
+            username=username,
+            email=email,
+        )
+        # todo: get course, where to get course?
+        course_id = ''
```

Comment:
[question] this should come from the LMS correct?

Reply:
yeah this would also have to come from the CSV. Which is just pulling this model https://github.com/openedx/edx-platform/blob/3a343f8301f71878628ad4e36e6160bdbec4862d/common/djangoapps/student/models/user.py#L1044